src/transactionsPage/utils.ts

```diff
@@ -0,0 +1,120 @@
+import { Transaction, Filters } from "./";
+import { SelectOptions } from "./filter";
+import { StatementStatistics } from "src/util/appStats";
+
+export const getAppNames = (transactions: Transaction[]) => {
+  return transactions.reduce(
+    (acc, current) => {
+      const twin = acc.some(
+        (item: SelectOptions) => item.label === current.stats_data.app,
+      );
+      return twin
+        ? acc
+        : acc.concat([
+            {
+              label: current.stats_data.app,
+              value: current.stats_data.app,
+            },
+          ]);
+    },
+    [{ label: "All", value: "All" }],
+  );
+};
+
+export const addTransactionStatements = (
+  transactions: Transaction[],
+  statements: StatementStatistics[],
+) => {
+  const data = transactions.map((item: Transaction) => {
+    const transactionStatements = item.stats_data.statement_ids.reduce(
+      (acc: string, current: string) => {
+        const statement = statements.find(
+          (stItem: StatementStatistics) => stItem.key.key_data.id === current,
+        );
+        // const newLine = idx > 0 ? "\n" : "";
+        return statement ? `${acc} ${statement.key.key_data.query}` : acc;
+      },
+      "",
+    );
+    return {
+      transactionStatements,
+      ...item,
+    };
+  });
+  return data;
+};
+
+export const collectStatementsText = (statements: StatementStatistics[]) => {
+  return statements.reduce(
```

Comment:
Why is this repeating code I see above? Don't have a problem with repetition if it's a different use case but I would expect the `transactionStatements` string computed above and this one to be the same for a given transaction.